### PR TITLE
No Cost filtering

### PR DESCRIPTION
**Pull request by kidManuel**
Small PR to allow adding no cost cards to a cost filter.

Before
Screenshot 2024-09-06 at 6 41 03 PM

After: 
Screenshot 2024-09-06 at 6 40 40 PM

### CODE REVIEW

File: src/store/lib/filtering.ts [hunk 1]
```diff
@@ -148,8 +148,12 @@ function filterOddCost(card: Card) {
...
 }
 
-function filterXCost(xCost: boolean) {
-  return (card: Card) => xCost && card.cost === -2;
+function filterXCost(card: Card) {
+  return card.cost === -2;
+}
+
+function filterNoCost(card: Card) {
+  return card.cost === undefined;
```
Comment (fspoettel) [inline src/store/lib/filtering.ts:156]: ArkhamDB sometimes has some (usually temporary until caught) inconsistencies with `null` and `undefined` in the JSON data, we should cover both `null` and `undefined` here to be safe.

```suggestion
  return card.cost == null;
```

File: src/components/filters/cost-filter.tsx [hunk 2]
```diff
@@ -120,6 +129,13 @@ export function CostFilter({ id }: { id: number }) {
           label={<i className="icon-x" />}
           onCheckedChange={onSetX}
         />
+        <Checkbox
```
Comment (fspoettel) [inline src/components/filters/cost-filter.tsx:132]: We could add `cols={2}` to the `<CheckboxGroup />` and re-order the  checkboxes like this? What do you think?

image

**Review by fspoettel — COMMENTED** [review]
Nice work! 🙌

File: src/store/lib/filtering.ts (hunk 1 above)
Comment (kidManuel) [reply src/store/lib/filtering.ts]: Sure! 
However, I am kind of a `==` hater. You have to think twice when using it, and it's the [reason why real programmers make fun of javascript](https://www.destroyallsoftware.com/talks/wat).

File: src/components/filters/cost-filter.tsx (hunk 2 above)
Comment (kidManuel) [reply src/components/filters/cost-filter.tsx:125]: hell yeah

**Comment by fspoettel** [comment]
Thank you! 👍

File: src/store/lib/filtering.ts (hunk 1 above)
Comment (fspoettel) [reply src/store/lib/filtering.ts]: @kidManuel so there is a lint for `===` against `==`, and in recent defaults, it explicitly ignores the case `== null` as it is actually useful.

Comment (fspoettel) [reply src/store/lib/filtering.ts]: The Wat talk is awesome (and valid), but I really don't think it applies here...

**Pull request merged**